web/src/test/java/org/cbioportal/web/SampleListControllerTest.java

```diff
@@ -0,0 +1,256 @@
+            Mockito.anyInt(), Mockito.anyString(), Mockito.anyString())).thenReturn(sampleLists);
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/sample-lists")
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.jsonPath("$", Matchers.hasSize(2)))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].listId").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].sampleListId").value(TEST_STABLE_ID_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].cancerStudyId").value(TEST_STUDY_IDENTIFIER_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].category").value(TEST_CATEGORY_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].name").value(TEST_NAME_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[0].description").value(TEST_DESCRIPTION_1))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].listId").doesNotExist())
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].sampleListId").value(TEST_STABLE_ID_2))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].cancerStudyId").value(TEST_STUDY_IDENTIFIER_2))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].category").value(TEST_CATEGORY_2))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].name").value(TEST_NAME_2))
+            .andExpect(MockMvcResultMatchers.jsonPath("$[1].description").value(TEST_DESCRIPTION_2));
+    }
+
+    @Test
+    public void getAllSampleListsMetaProjection() throws Exception {
+
+        BaseMeta baseMeta = new BaseMeta();
+        baseMeta.setTotalCount(2);
+
+        Mockito.when(sampleListService.getMetaSampleLists()).thenReturn(baseMeta);
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/sample-lists")
+            .param("projection", "META"))
+            .andExpect(MockMvcResultMatchers.status().isOk())
+            .andExpect(MockMvcResultMatchers.header().string(HeaderKeyConstants.TOTAL_COUNT, "2"));
+    }
+
+    @Test
+    public void getSampleListNotFound() throws Exception {
+
+        Mockito.when(sampleListService.getSampleList(Mockito.anyString())).thenThrow(
+            new SampleListNotFoundException("test_sample_list_id"));
+
+        mockMvc.perform(MockMvcRequestBuilders.get("/sample-lists/test_sample_list_id")
+            .accept(MediaType.APPLICATION_JSON))
+            .andExpect(MockMvcResultMatchers.status().isNotFound())
+            .andExpect(MockMvcResultMatchers.jsonPath("$.message")
+                .value("Sample list not found: test_sample_list_id"));
+    }
+
+    @Test
+    public void getSampleList() throws Exception {
+
+        SampleList sampleList = new SampleList();
+        sampleList.setListId(TEST_LIST_ID_1);
```

Comment:
Actually this is how I normally do it too. I think manual code-based mocks are good, I haven't seen a cleaner way of doing this, this is quite clear.